jnigen/example/in_app_java/lib/android_utils.dart

```diff
@@ -36,6 +38,17 @@ class AndroidUtils extends jni.JniObject {
       _showToast(mainActivity.reference, text.reference, duration).check();
 }
 
+extension AndroidUtilsJniArray on jni.JniArray<AndroidUtils> {
+  AndroidUtils operator [](int index) {
+    return AndroidUtils.fromRef(
+        elementAt(index, jni.JniType.objectType).object);
```

Comment:
Maybe elementAt can call `object` getter? Are you explicitly checking `JniResult.exception` somewhere?

Reply:
I'm also using the same `elementAt` for primitives, I could have a `_primitiveElementAt` for those and another `elementAt(index)` that I expose for other classes to use. The problem with that is you could mistakenly write `primitiveArray.elementAt(3)` and get an error and not know why!